### PR TITLE
Add German translation

### PR DESCRIPTION
**Pull request by stefanreimers**

### CODE REVIEW

**Comment by mrstegeman** [comment]
@tim-hellhake would you be interested in taking a look at this?

File: static/fluent/de/main.ftl [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+# Wordmark
+wordmark =
+  .alt = Mozilla IoT
+
+## Menu Items
+assistant-menu-item = Assistent
+things-menu-item = Dinge
+rules-menu-item = Regeln
+logs-menu-item = Protokolle
+floorplan-menu-item = Lageplan
```
Comment (c727) [inline static/fluent/de/main.ftl]: Raumplan

File: static/fluent/de/main.ftl [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+settings-menu-item = Einstellungen
+log-out-button = Ausloggen
+
+## Assistant
+assistant-avatar-image =
+  .alt = Assistent Avatar
```
Comment (c727) [inline static/fluent/de/main.ftl]: Assistent-Avatar

File: static/fluent/de/main.ftl [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+assistant-controls-text-input =
+  .placeholder = Wie kann ich helfen?
+
+## Floorplan
+upload-floorplan = Lageplan aktualisieren...
```
Comment (c727) [inline static/fluent/de/main.ftl]: Raumplan

File: static/fluent/de/main.ftl [hunk 4]
```diff
@@ -0,0 +1,526 @@
...
+network-settings-wifi-image =
+  .alt = Kabellose Verbindung
+network-settings-wifi = Kabellose Verbindung
+network-settings-home-network-image =
+  .alt = Heimnetzwerk
+network-settings-internet-image =
+  .alt = Internet
+network-settings-configure = Konfigurieren
+network-settings-internet-wan = Internet (WAN)
+network-settings-home-network-lan = Heimneztzwerk (LAN)
```
Comment (c727) [inline static/fluent/de/main.ftl]: -z

File: static/fluent/de/main.ftl [hunk 5]
```diff
@@ -0,0 +1,526 @@
...
+network-settings-wifi-wlan = Wi-Fi (WLAN)
+network-settings-ip-address = IP Adresse
```
Comment (c727) [inline static/fluent/de/main.ftl]: IP-Adresse

File: static/fluent/de/main.ftl [hunk 6]
```diff
@@ -0,0 +1,526 @@
...
+network-settings-dhcp = Automatisch (DHCP)
+network-settings-static = Manuell (Statische IP)
+network-settings-pppoe = Bridge (PPPoE)
+network-settings-static-ip-address = Statische IP Adresse
```
Comment (c727) [inline static/fluent/de/main.ftl]: IP-Adresse

File: static/fluent/de/main.ftl [hunk 7]
```diff
@@ -0,0 +1,526 @@
...
+network-settings-network-mask = Netzwerkmaske
+network-settings-gateway = Gateway
+network-settings-done = Erledigt
+network-settings-wifi-password =
+  .placeholder = Passwort
+network-settings-show-password = Passwort anzeigen
+network-settings-connect = Verbinden
+network-settings-username = Benutzername
+network-settings-password = Passwort
+network-settings-router-ip = Router IP Adresse
```
Comment (c727) [inline static/fluent/de/main.ftl]: Router-IP-Adresse

File: static/fluent/de/main.ftl [hunk 8]
```diff
@@ -0,0 +1,526 @@
...
+network-settings-dhcp-server = DHCP Server
```
Comment (c727) [inline static/fluent/de/main.ftl]: DHCP-Server

File: static/fluent/de/main.ftl [hunk 9]
```diff
@@ -0,0 +1,526 @@
...
+network-settings-enable-wifi = Wi-Fi aktivieren
+network-settings-network-name = Netwerkname (SSID)
```
Comment (c727) [inline static/fluent/de/main.ftl]: Netzwerkname

File: static/fluent/de/main.ftl [hunk 10]
```diff
@@ -0,0 +1,526 @@
...
+wireless-connected = Verbunden
+wireless-icon =
+  .alt = Wi-Fi Netzwerk
```
Comment (c727) [inline static/fluent/de/main.ftl]: Wi-Fi-Netzwerk

File: static/fluent/de/main.ftl [hunk 11]
```diff
@@ -0,0 +1,526 @@
...
+rules-delete-icon =
+  .alt = Löschen
+rules-drag-hint = Gerät hierher ziehen, um eine Regel zu erzeugen
+rules-drag-input-hint = Gerät als Eingang hinzufügen
+rules-drag-output-hint = Gerät als Ausgang hinzufügen
+rules-scroll-left =
+  .alt = Links scrollen
+rules-scroll-right =
+  .alt = Rechts scrollen
+rules-delete-prompt = Gerät hier fallen lassen um es zu  löschen
```
Comment (c727) [inline static/fluent/de/main.ftl]: ... lassen, um es zu löschen

File: static/fluent/de/main.ftl [hunk 12]
```diff
@@ -0,0 +1,526 @@
...
+add-thing-done = Erledigt
+add-thing-cancel = Abbrechen
+
+## Context Menu
+context-menu-choose-icon = Bild wählen...
+context-menu-save = Speichern
+context-menu-remove = Löschen
+
+## Capabilities
+OnOffSwitch = Ein/Aus Schalter
```
Comment (c727) [inline static/fluent/de/main.ftl]: Ein/Aus-Schalter

File: static/fluent/de/main.ftl [hunk 13]
```diff
@@ -0,0 +1,526 @@
...
+MultiLevelSwitch = Mehrfachschalter
+ColorControl = Farbsteuerung
+EnergyMonitor = Energie-Monitor
+BinarySensor = Binärer Sensor
+MultiLevelSensor = Mehrfachssensor
```
Comment (c727) [inline static/fluent/de/main.ftl]: -s

File: static/fluent/de/main.ftl [hunk 14]
```diff
@@ -0,0 +1,526 @@
...
+on = Ein
+off = Aus
+power = Strom
+voltage = Spannung
+temperature = Temperatur
+current = Aktuell
+frequency = Frequenz
+color = Farbe
+brightness = Helligkeit
+leak = Leck
```
Comment (c727) [inline static/fluent/de/main.ftl]: Without checking the english version i think we need an adjective here, like "Undicht"/"Feucht"

File: static/fluent/de/main.ftl [hunk 15]
```diff
@@ -0,0 +1,526 @@
...
+## Domain Setup
+reclaim-prompt = Die Subdomain scheint bereits registriert zu sein. Zum Wiederherstellen
+click-here = hier klicken
+check-email-for-token = Bitte das Reclaim-Token aus der verschickten E-Mail oben einfügen.
+reclaim-failed = Subdomain konnte nicht wiederhergestellt werden.
+subdomain-already-used = Diese Subdomain ist bereits in Gebrauch. Bitte einen anderen Namen wählen.
+invalid-reclamation-token = Ungültiges Reclaim-Token.
+domain-success = Erfolgreich! Es wird weitergeleitet...
+issuing-error = Zertifikat konnte nicht erstellt werden. Bitte erneut versuchen.
+redirecting = Redirecting...
```
Comment (c727) [inline static/fluent/de/main.ftl]: Weiterleiten...

File: static/fluent/de/main.ftl [hunk 16]
```diff
@@ -0,0 +1,526 @@
...
+addon-discovery-added = Hinzugefügt
+addon-discovery-add = Hinzufügen
+addon-discovery-installing = Installieren...
+addon-discovery-failed = Fehlgeschlagen
+
+## Page Titles
+settings = Einstellungen
+domain = Domäne
+users = Benutzer
+edit-user = Benutzer anpassen
```
Comment (c727) [inline static/fluent/de/main.ftl]: bearbeiten

File: static/fluent/de/main.ftl [hunk 17]
```diff
@@ -0,0 +1,526 @@
...
+ { $gateway-link } getätigt werden.
+
+## Creating Wi-Fi Network Page
+creating-title = Neues Wi-Fi Netzwerk anlegen - WebThings Gateway
+creating-header = Neues Wi-Fi Netzwerk wird anlegt...
+creating-content = Bitte Verbindung mit dem WLAN { $ssid } mit dem gerade vergebenen Passwort 
+ aufbauen und im Webbrowser entweder { $gateway-link } oder { $ip-link } starten.
+
+## General Terms
+ok = Ok
```
Comment (c727) [inline static/fluent/de/main.ftl]: OK

File: static/fluent/de/main.ftl [hunk 18]
```diff
@@ -0,0 +1,526 @@
...
+
+## Speech
+speech-unsupported = Der Browser unterstützt keine Sprachausgabe
+speech-didnt-get = Sorry, das habe ich nicht verstanden.
+
+## Errors
+unknown-state = Unbekannter Zustand.
+error = Fehler
+errors = Fehler
+gateway-unreachable = Gateway unerreichbar
```
Comment (c727) [inline static/fluent/de/main.ftl]: nicht erreichbar

File: static/fluent/de/main.ftl [hunk 19]
```diff
@@ -0,0 +1,526 @@
...
+more-information = Mehr Informationen
+invalid-file = Ungültige Datei.
+failed-read-file = Datei nicht lesbar.
+failed-save = Fehler beim Speichern.
+
+## Schema Form
+unsupported-field = Feldschema nicht untertützt
+
+## Icon Sources
```
Comment (c727) [inline static/fluent/de/main.ftl]: Ist this needed in Translation?

File: static/fluent/de/main.ftl [hunk 20]
```diff
@@ -0,0 +1,526 @@
...
+signup-password-mismatch = Passwörter sind nicht gleich
+signup-next = Weiter
+
+## Tunnel Setup Page
+tunnel-setup-title = Subdomain wählen - WebThings Gateway
+tunnel-setup-welcome = Willkommen
+tunnel-setup-choose-address = Sichere URL für das Gateway:
+tunnel-setup-input-subdomain =
+  .placeholder = Subdomain
+tunnel-setup-opt-in = Benachrichtigungen bei neuen Funktionen und Möglichkiten zunm Mitmachen erhalten.
```
Comment (c727) [inline static/fluent/de/main.ftl]: Möglichkeiten zum

File: static/fluent/de/main.ftl [hunk 21]
```diff
@@ -0,0 +1,526 @@
...
+tunnel-setup-create = Erzeugen
+tunnel-setup-skip = Überspringen
+
+## Authorize Page
+authorize-title = Authorisierungsanfrage - WebThings Gateway
+authorize-authorization-request = Authorisierungsanfrage
+authorize-would-like = möchte auf das Gateway zugreifen zum 
+authorize-devices = Geräte.
+authorize-from = von
+authorize-monitor-and-control = Übewachen und kontrollieren
```
Comment (c727) [inline static/fluent/de/main.ftl]: Überwachen

File: static/fluent/de/main.ftl [hunk 22]
```diff
@@ -0,0 +1,526 @@
...
+authorize-allow =
+  .value = Erlauben
+authorize-deny = Ablehnen
+
+## Local Token Page
+local-token-title = Dienst für lokale Token - WebThings Gateway
+local-token-header = Dienst für lokale Token
+local-token-your-token = Das lokale Zugriffstoken ist
+local-token-jwt = JSON Web Token
+local-token-use-it = Zuglriff auf das Gateway über
```
Comment (c727) [inline static/fluent/de/main.ftl]: Zugriff

File: static/fluent/de/main.ftl [hunk 23]
```diff
@@ -0,0 +1,526 @@
...
+local-token-bearer-type = Bearer-type Authorisierung
+
+## Router Setup Page
+router-setup-title = Router Setup - WebThings Gateway
```
Comment (c727) [inline static/fluent/de/main.ftl]: Router-Setup

File: static/fluent/de/main.ftl [hunk 24]
```diff
@@ -0,0 +1,526 @@
...
+router-setup-header = Neues Wi-Fi Netzwerk erzeugen
```
Comment (c727) [inline static/fluent/de/main.ftl]: Wi-Fi-Netzwerk

File: static/fluent/de/main.ftl [hunk 25]
```diff
@@ -0,0 +1,526 @@
...
+router-setup-input-password =
+  .placeholder = Passwort
+router-setup-input-confirm-password =
+  .placeholder = Passwort bestätigen
+router-setup-create =
+  .value = Anlegen
+router-setup-password-mismatch = Passwörter sind nicht gleich
+
+## Wi-Fi Setup Page
+wifi-setup-title = Wi-Fi Setup - WebThings Gateway
```
Comment (c727) [inline static/fluent/de/main.ftl]: Wi-Fi-Setup

File: static/fluent/de/main.ftl [hunk 26]
```diff
@@ -0,0 +1,526 @@
...
+wifi-setup-header = Mit Wi-Fi Netzwerk verbinden?
```
Comment (c727) [inline static/fluent/de/main.ftl]: Minus

File: static/fluent/de/main.ftl [hunk 27]
```diff
@@ -0,0 +1,526 @@
...
+wifi-setup-input-password =
+  .placeholder = Passwort
+wifi-setup-show-password = Passwort anzeigen
+wifi-setup-connect =
+  .value = Verbinden
+wifi-setup-network-icon =
+  .alt = Wi-Fi Netzwerk
```
Comment (c727) [inline static/fluent/de/main.ftl]: Minus

File: static/fluent/de/main.ftl [hunk 28]
```diff
@@ -0,0 +1,526 @@
...
+wifi-setup-skip = Überspringen
+
+## Connecting to Wi-Fi Page
+connecting-title = Wi-Fi Verbindung herstellen - WebThings Gateway
```
Comment (c727) [inline static/fluent/de/main.ftl]: Minus

File: static/fluent/de/main.ftl [hunk 29]
```diff
@@ -0,0 +1,526 @@
...
+connecting-header = Verbindung wird hergestellt...
+connecting-connect = Bitte mit demselben Netzwerk wie das Gateway verbinden und 
+ das Setup unter { $gateway-link } im Webbrowser fortsetzen.
+connecting-warning = Hinweis: Wenn { $domain } nicht erreichbar ist, kann die IP Adresse
```
Comment (c727) [inline static/fluent/de/main.ftl]: Minus

File: static/fluent/de/main.ftl [hunk 30]
```diff
@@ -0,0 +1,526 @@
...
+ des Gateways in den Routereinstellungen nachgeschlagen werden.
+connecting-header-skipped = Wi-Fi Setup abgebrochen
```
Comment (c727) [inline static/fluent/de/main.ftl]: Minus

File: static/fluent/de/main.ftl [hunk 31]
```diff
@@ -0,0 +1,526 @@
...
+connecting-skipped = Das Gateway ist einsatzbereit. Einstellungen können unter
+ { $gateway-link } getätigt werden.
+
+## Creating Wi-Fi Network Page
+creating-title = Neues Wi-Fi Netzwerk anlegen - WebThings Gateway
+creating-header = Neues Wi-Fi Netzwerk wird anlegt...
```
Comment (c727) [inline static/fluent/de/main.ftl]: Minus, angelegt

File: static/fluent/de/main.ftl [hunk 32]
```diff
@@ -0,0 +1,526 @@
...
+connecting-connect = Bitte mit demselben Netzwerk wie das Gateway verbinden und 
+ das Setup unter { $gateway-link } im Webbrowser fortsetzen.
+connecting-warning = Hinweis: Wenn { $domain } nicht erreichbar ist, kann die IP Adresse
+ des Gateways in den Routereinstellungen nachgeschlagen werden.
+connecting-header-skipped = Wi-Fi Setup abgebrochen
+connecting-skipped = Das Gateway ist einsatzbereit. Einstellungen können unter
+ { $gateway-link } getätigt werden.
+
+## Creating Wi-Fi Network Page
+creating-title = Neues Wi-Fi Netzwerk anlegen - WebThings Gateway
```
Comment (c727) [inline static/fluent/de/main.ftl]: Minus

File: static/fluent/de/main.ftl [hunk 33]
```diff
@@ -0,0 +1,526 @@
...
+creating-header = Neues Wi-Fi Netzwerk wird anlegt...
+creating-content = Bitte Verbindung mit dem WLAN { $ssid } mit dem gerade vergebenen Passwort 
+ aufbauen und im Webbrowser entweder { $gateway-link } oder { $ip-link } starten.
```
Comment (c727) [inline static/fluent/de/main.ftl]: aufrufen

File: static/fluent/de/main.ftl (hunk 17 above)
Comment (stefanreimers) [reply static/fluent/de/main.ftl]: Incorporated

File: static/fluent/de/main.ftl (hunk 31 above)
Comment (stefanreimers) [reply static/fluent/de/main.ftl]: Incorporated

**Comment by stefanreimers** [comment]
@c727 Thanks for the review. I have incorporated the changes. 2 changes are left (line 267 / 421) and open to discussion.
Regarding the "x minutes ago" => "vor x Minuten": Does fluent support placeholders in translations, like  "vor %s Minuten"?

**Comment by tim-hellhake** [comment]
@mrstegeman sure!

**Comment by mrstegeman** [comment]
> @c727 Thanks for the review. I have incorporated the changes. 2 changes are left (line 267 / 421) and open to discussion.
> Regarding the "x minutes ago" => "vor x Minuten": Does fluent support placeholders in translations, like "vor %s Minuten"?

This will be supported once #2284 is merged.

File: static/fluent/de/main.ftl [hunk 34]
```diff
@@ -0,0 +1,526 @@
...
+
+## Top-Level Settings
+settings-domain = Domäne
+settings-network = Netzwerk
+settings-users = Benutzer
+settings-add-ons = Erweiterungen
+settings-adapters = Adapter
+settings-localization = Lokalisierung
+settings-updates = Aktualisierungen
+settings-authorizations = Authorisierungen
```
[the remark this answers is not in the document]
Comment (stefanreimers) [reply static/fluent/de/main.ftl]: ok

File: static/fluent/de/main.ftl [hunk 35]
```diff
@@ -0,0 +1,526 @@
...
+  .placeholder = Gateway
+
+## Network Settings
+network-settings-unsupported = Netzwerkeinstellungen werden auf dieser Plattform nicht unterstützt.
+network-settings-ethernet-image =
+  .alt = Kabelverbindung
+network-settings-ethernet = Kabelverbindung
+network-settings-wifi-image =
+  .alt = Kabellose Verbindung
+network-settings-wifi = Kabellose Verbindung
```
[the remark this answers is not in the document]
Comment (stefanreimers) [reply static/fluent/de/main.ftl]: ok

File: static/fluent/de/main.ftl [hunk 36]
```diff
@@ -0,0 +1,526 @@
...
+  .placeholder = Email
+user-settings-input-password =
+  .placeholder = Passwort
+user-settings-input-new-password =
+  .placeholder = Neues Passwort (Optional)
+user-settings-input-confirm-new-password =
+  .placeholder = Neues Passwort bestätigen
+user-settings-input-confirm-password =
+  .placeholder = Passwort bestätigen
+user-settings-password-mismatch = Passwörter sind nicht gleich
```
[the remark this answers is not in the document]
Comment (stefanreimers) [reply static/fluent/de/main.ftl]: ok

File: static/fluent/de/main.ftl [hunk 37]
```diff
@@ -0,0 +1,526 @@
...
+current-version = Aktuelle Version
+failed = Fehlgeschlagen
+never = Niemals
+in-progress = Laufend
+restarting = Neustarten
+
+## Developer Settings
+developer-settings-enable-ssh = SSH aktivieren
+developer-settings-view-internal-logs = Interne Protokolle ansehen
+developer-settings-create-local-authorization = Lokale Authorisierung erstellen
```
[the remark this answers is not in the document]
Comment (stefanreimers) [reply static/fluent/de/main.ftl]: ok

File: static/fluent/de/main.ftl [hunk 38]
```diff
@@ -0,0 +1,526 @@
...
+Custom = Benutzerdefiniertes Gerät
+
+## Properties
+alarm = Alarm
+pushed = Gedrückt
+not-pushed = Nicht gedrückt
+on-off = Ein/Aus
+on = Ein
+off = Aus
+power = Strom
```
[the remark this answers is not in the document]
Comment (stefanreimers) [reply static/fluent/de/main.ftl]: ok

File: static/fluent/de/main.ftl [hunk 39]
```diff
@@ -0,0 +1,526 @@
...
+voltage = Spannung
+temperature = Temperatur
+current = Aktuell
```
[the remark this answers is not in the document]
Comment (stefanreimers) [reply static/fluent/de/main.ftl]: Good to know, didn't know that in electric context. I think _Stromstärke_ then might be correct.

File: static/fluent/de/main.ftl [hunk 40]
```diff
@@ -0,0 +1,526 @@
...
+authorizations = Authorisierungen
+developer = Entwickler
+network = Netzwerk
+ethernet = Ethernet
+wifi = Wi-Fi
+icon = Bild
+
+## Speech
+speech-unsupported = Der Browser unterstützt keine Sprachausgabe
+speech-didnt-get = Sorry, das habe ich nicht verstanden.
```
[the remark this answers is not in the document]
Comment (stefanreimers) [reply static/fluent/de/main.ftl]: ok

File: static/fluent/de/main.ftl [hunk 41]
```diff
@@ -0,0 +1,526 @@
...
+settings-localization = Lokalisierung
+settings-updates = Aktualisierungen
+settings-authorizations = Authorisierungen
+settings-experiments = Experimente
+settings-developer = Entwickler
+
+## Domain Settings
+domain-settings-local-label = Lokaler Zugriff
+domain-settings-local-update = Hostnamen aktualisieren
+domain-settings-remote-access = Zugriff übers Web
```
[the remark this answers is not in the document]
Comment (stefanreimers) [reply static/fluent/de/main.ftl]: ok

**Comment by mrstegeman** [comment]
Please rebase on the latest master and check out the changes to the English translation.

**Comment by mrstegeman** [comment]
@stefanreimers I've rebased for you and updated a few things. The only changes that need to be made are:
1. Add missing `tunnel-setup-time-sync` translation.
2. Update the `utils-X-ago` translations to use the CLDR plurals.

**Comment by mrstegeman** [comment]
@stefanreimers you can now find this project in Pontoon: https://pontoon.mozilla.org/projects/webthings-gateway/

**Comment by c727** [comment]
@mrstegeman I updated some German translations at Pontoon...

**Pull request merged**